pkg/kritis/metadata/containeranalysis/containeranalysis_int_test.go

```diff
@@ -137,3 +140,30 @@ func TestCreateAttestationNoteAndOccurrence(t *testing.T) {
 		}
 	}
 }
+
+func TestGetMultiplePages_Vulnerabilities(t *testing.T) {
+	d, err := New()
+	if err != nil {
+		t.Fatalf("Could not initialize the client %s", err)
+	}
+
+	// Set PageSize to 300
+	createListOccurrencesRequest = createListOccurrencesRequest_test
+
+	vuln, err := d.Vulnerabilities("gcr.io/kritis-int-test/java-with-vulnz@sha256:358687cfd3ec8e1dfeb2bf51b5110e4e16f6df71f64fba01986f720b2fcba68a")
+	if err != nil {
+		t.Fatalf("Found err %s", err)
+	}
+
+	if len(vuln) <= 300 {
```

Comment:
With paging APIs, you aren't actually guaranteed to receive the full page of results, up to the max number you set. Therefore, you can't rely on this as a signal for whether there are more results. Instead, you'd want to check whether the next page token is set, to figure out if more pages of results are available.

Reply:
I don't actually fully understand what @aysylu meant here.

Apart from that, I think a more rigorous test is to check if len(vulnz) > 600, so we know for sure it goes over at least two pages.

Reply:
In response to @aysylu,
I think my reasoning for this assertion is that `vuln` should contain all of the Vulnerabilities, and if the length of `vuln` exceeds the page size, then we must have extracted results from multiple pages. So it must have implicitly set the next page token.

I can try more directly testing whether the next page token is set, if you think that would be a better approach

Reply:
In response to @ooq,
For a stronger assertion, I modified the page size to 100 and checked if len(vulnz) > 900. The page size is still large enough to not time out.

Reply:
Re API for pagination, best practices: https://aip.dev/158 "The API may return fewer results than the number requested (including zero results), even if not at the end of the collection."

Reply:
I see, so I'd like to test that the next page token is being set, but none of our code actually does that. That's happening within the Container Analysis API library: https://github.com/grafeas/kritis/blob/588f70e090ad2ee4c98e65a6d105976ade4aabb2/vendor/cloud.google.com/go/containeranalysis/apiv1beta1/grafeas_v1_beta1_client.go#L188 

Our code enters an infinite loop until the client library to returns iterator.Done: https://github.com/grafeas/kritis/blob/588f70e090ad2ee4c98e65a6d105976ade4aabb2/pkg/kritis/metadata/containeranalysis/containeranalysis.go#L119.

Should I revise the test to make sure that we are receiving the iterator.Done signal?